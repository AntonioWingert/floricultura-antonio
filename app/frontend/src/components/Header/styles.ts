import theme from '../../styles/theme';
import tw from 'tailwind-styled-components';

export const Container = tw.header`
hidden
sm:flex
sm:w-screen
sm:justify-between
sm:items-center
sm:h-28
sm:${() => theme['padding-sizes'].xxxl}
sm:${() => theme['background-colors'].primary}

`;
