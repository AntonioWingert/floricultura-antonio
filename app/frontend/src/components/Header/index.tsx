import React from 'react';
import * as S from './styles';

import CustomTitle from '../CustomTitle';
import Image from 'next/image';
import User from '../../../public/user.svg';

export default function Header({ children }: { children: React.ReactNode }) {
  return (
    <S.Container>
      <CustomTitle $color="dark" $size="xxxl">
        {children}
      </CustomTitle>
      <Image src={User} alt="User Icon" className="cursor-pointer" />
    </S.Container>
  );
}
