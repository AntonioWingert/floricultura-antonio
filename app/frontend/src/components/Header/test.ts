import { render, screen } from '@testing-library/react';
import Header from '.';

describe('<Header />', () => {
  it('should render the heading', () => {
    const { container } = render(Header({ children: 'Header' }));
    expect(screen.getByRole('heading', { name: /Header/i })).toBeVisible();
    expect(container.firstChild).toMatchSnapshot();
  });
});
