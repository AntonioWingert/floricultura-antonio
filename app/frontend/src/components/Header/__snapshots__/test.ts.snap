// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the heading 1`] = `
<header
  class="hidden sm:flex sm:w-screen sm:justify-between sm:items-center sm:h-28 sm: p-12 sm: bg-blue-300 dark:bg-blue-800"
>
  <h1
    class="text-gray-800 text-3xl text-center"
  >
    Header
  </h1>
  <img
    alt="User Icon"
    class="cursor-pointer"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style="color: transparent;"
    width="40"
  />
</header>
`;
